sympy/solvers/ode/tests/test_single.py

```diff
@@ -50,7 +50,7 @@
 import traceback
 
 
-x = Symbol('x')
+x = Symbol('x', real=True)
```

Comment:
I don't think we should change this globally for all tests in this file. It would be better to have a new symbol like:
```
xr = Symbol('x', real=True)
```
Then that could be used only in the tests where it is needed. If we need to use `xr` for a test then there should also be a test that shows what happens with `x`.

Reply:
It's also possible to do
```
fr = Function('f', real=True)
```
if that makes a difference.

Reply:
both approach still fails for checkodesol flag. so should i leave the new flag as it is and change the global `x=Symbol('x')`?

Reply:
Yes